core/kernel/persistence/smoothsql/install/SmoothRdsModel.php

```diff
@@ -17,43 +17,45 @@
  * Copyright (c) 2018 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
  *
  */
-namespace   oat\generis\model\kernel\persistence\smoothsql\install;
+
+namespace oat\generis\model\kernel\persistence\smoothsql\install;
 
 use Doctrine\DBAL\Schema\Schema;
+
 /**
  * Helper to setup the required tables for generis smoothsql
  */
-class SmoothRdsModel {
-
+class SmoothRdsModel
+{
     /**
-     * 
+     *
      * @param Schema $schema
+     *
      * @return \Doctrine\DBAL\Schema\Schema
      */
     public static function addSmoothTables(Schema $schema)
     {
-        $table = $schema->createTable("models");
-        $table->addColumn('modelid', "integer",array("notnull" => true,"autoincrement" => true));
-        $table->addColumn('modeluri', "string", array("length" => 255,"default" => null));
-        $table->addOption('engine' , 'MyISAM');
-        $table->setPrimaryKey(array('modelid'));
-
-        $table = $schema->createTable("statements");
-        $table->addColumn("modelid", "integer",array("notnull" => true,"default" => 0));
-        $table->addColumn("subject", "string",array("length" => 255,"default" => null));
-        $table->addColumn("predicate", "string",array("length" => 255,"default" => null));
-        $table->addColumn("object", "text", array("default" => null,"notnull" => false));
-            
-        $table->addColumn("l_language", "string",array("length" => 255,"default" => null,"notnull" => false));
-        $table->addColumn("id", "integer",array("notnull" => true,"autoincrement" => true));
-        $table->addColumn("author", "string",array("length" => 255,"default" => null,"notnull" => false));
-        $table->setPrimaryKey(array("id"));
-        $table->addOption('engine' , 'MyISAM');
-        $table->addColumn("epoch", "string" , array("notnull" => null));
-            
-        $table->addIndex(array("subject","predicate"),"k_sp");
-        $table->addIndex(array("predicate","object"),"k_po");
-        
+        $table = $schema->createTable('models');
+        $table->addColumn('modelid', 'string', ['length' => 25, 'notnull' => true]);
+        $table->addColumn('modeluri', 'string', ['length' => 255]);
```

Comment:
No I didn't check. In fact, this shouldn't have been changed at all since gcp Spanner asks for some specific types, and especially doesn't support auto-increment. Also, the scope of the lib-dbal-spanner excluded the schema management so for the moment, the benchmark with spanner will require a manual table creation. I will amend this to make it compatible for mysql.

Reply:
Forget my previous comment.
I added the engine option back.
I did check the whole cycle and it works.